package.json

```diff
@@ -125,7 +125,7 @@
     "ua-parser-js": "^0.7.22"
   },
   "scripts": {
-    "preinstall": "test -e ./autogen.sh && ./autogen.sh --prefix=/usr --datarootdir=/share || :",
+    "preinstall": "scripts/yarn-preinstall.sh",
```

Comment:
can we instead check for artifacts generated by `autogen.sh` ?

Reply:
TL;DR - No. There is deep dark magic going on here!

Basically, we need a balance about where and how to run autogen.sh in a number of different scenarios...
  - as a developer running `yarn install` after a repo clone or rebase
  - `yarn install` in ovirt-engine-nodejs-modules where only package.json and yarn.lock are available
  - under `make` either manually or by the build system

So, `autogen.sh` may not be available / not needed (nodejs-modules) or we always want to run it (developer) or we run it external to `yarn install` and can skip running it (build system).  It's all dark magic.

For the developer scenario, we only need to config files to be substituted in by autogen.  See:
https://github.com/oVirt/ovirt-web-ui/blob/4aee8f2646ed05296dda2bd164a6582edcaeb6b9/configure.ac#L40-L44

Moving it to a regular shell script makes it much easier to control when a `yarn install` will call `autogen.sh`.

Reply:
OK. Please describe the reasons (mentioned above) in a comment in `yarn-preinstall.sh`

Reply:
I had to add the `test -e` back to the package.json script to support inclusion in nodejs-modules, but left the rest of the code as-is.  Added a big fat comment in yarn-preinstall.sh.